backend/src/email/routes/email-campaign.routes.ts

```diff
@@ -88,6 +88,16 @@ const duplicateCampaignValidator = {
   }),
 }
 
+const tesseractCampaignValidator = {
+  [Segments.BODY]: Joi.object({
+    url: Joi.string()
+      .pattern(/vault.gov.sg/)
+      .uri({ scheme: 'https' })
+      .trim()
+      .required(),
+  }),
```

Comment:
Not sure if this is a well defined validator? I also check for valid vault domain in the `isValidVaultUrl` middleware

Reply:
Another way this can be done is to implement a custom `Joi` validator. See [fromAddressValidator](https://github.com/opengovsg/postmangovsg/blob/6869ac10e933c13df9b069ae54c0478d0154aa55/backend/src/core/utils/from-address.ts#L33) for an example.